File: Graph/C#/2331 반복수열.cs
using System;
using System.Collections;


class Program
{
    int cal(int num, int _p) {
       
       int result = 0;
       int n = 0;
       while(num != 0) {
           n = num%10;
           n = (int)Math.Pow(n, _p);
           result += n;
           num = num/10;
       }
       return result;
    }
    
    static void Main() {
        string s = Console.ReadLine();
        string[] num = s.Split(" ");
        ArrayList arr = new ArrayList();
        bool[] visit = new bool[250000];
        int a = int.Parse(num[0]);
        int p = int.Parse(num[1]);
        
        arr.Add(a);
        Console.WriteLine(arr[0]);
        Console.WriteLine(cal(arr[0], p));
        
        
    }
}

File: _etc/Java/2331 반복수열.java
import java.util.Scanner;
import java.util.*;


public class Main {
    static int a,p;
    
    static int cal(int num) {
        int result = 0;
        int temp = 0;
        while(num != 0) {
            temp = num%10;
            temp = (int)Math.pow(temp, p);
            result += temp;
            num /= 10;
        }
        
        return result;
    }
    
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        a = sc.nextInt();
        p = sc.nextInt();
        ArrayList<Integer> arr = new ArrayList<Integer>();
        int check = -1;
        arr.add(a);
        
        while(true) {
            int v = cal(arr.get(arr.size()-1));
            
            if(arr.contains(v)) {
                check = v;
                break;
            }
            arr.add(v);
        }
        
        
        
        System.out.printf("%d", arr.indexOf(check));
    }
}
